app/models/instance.rb

```diff
@@ -15,6 +15,9 @@ def find_tenant_by_host(host)
     end
   end
 
+  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
+  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
```

Comment:
Or you can use Ruby mine to see where it is defined.
On 8 Jan, 2015 6:10 pm, "Joel Low" <notifications@github.com> wrote:

> In app/models/instance.rb
> <https://github.com/Coursemology/coursemology2/pull/41#discussion-diff-22643686>
> :
>
> > @@ -15,6 +15,9 @@ def find_tenant_by_host(host)
> >      end
> >    end
> >
> > +  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
> > +  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
>
> modules are split up, like models/controllers/views.
>
> actually, i still think the simplest is just trace into the function call
> in a debugger lah. it's not hard to put breakpoint and step-into...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Coursemology/coursemology2/pull/41/files#r22643686>.
>